MedicalDocManagment.WebUI/app/users-search-form.component.ts

```diff
@@ -0,0 +1,92 @@
+﻿import { Component } from '@angular/core';
+import { NgForm } from '@angular/forms';
+import { Response } from '@angular/http';
+
+import { AdminService } from './admin.service';
+import UserModel from './models/user.model';
+import UsersModel from './models/users.model';
+import UserSearchModel from './models/user-search.model';
+import { UsersSearchOptionsEnum } from './users-search-options.enum';
+
+@Component({
+  moduleId: module.id,
+  selector: 'users-search-Form',
+  templateUrl: 'views/users-search-form.component.html',
+  providers: [AdminService]
+})
+
+export class UsersSearchFormComponent {
+  // next is declared in such way so that enum can be used in template
+  UsersSearchOptionsEnum = UsersSearchOptionsEnum;
+
+  private _isErrorOnSearching: boolean;
+  private _isSearching: boolean;
+  private _adminService: AdminService;
+  private _lastErrorMessage: string;
+  private _searchOption: UsersSearchOptionsEnum;
+  private _searchResult: UsersModel;
+  private  _userToSearchFor: UserSearchModel;
+
+  constructor(adminService: AdminService) {
+    this._isErrorOnSearching = false;
+    this._isSearching = false;
+    this._lastErrorMessage = '';
+    this._searchResult = null;
+    this._searchOption = UsersSearchOptionsEnum.byUsername;
+    this._userToSearchFor = new UserSearchModel();
+
+    this._adminService = adminService;
+  }
+
+  search(): void {
+    this._isSearching = true;
+    this._isErrorOnSearching = false;
+    this._searchResult = null;
+    switch (this._searchOption) {
+      case UsersSearchOptionsEnum.byPositionName:
+        this._searchByPositionName();
+        break;
+      case UsersSearchOptionsEnum.byStatus:
+        this._searchByStatus();
+        break;
+      case UsersSearchOptionsEnum.byUsername:
+        this._searchByUsername();
+        break;
+    }
+  }
+
+  private _handleSearchError(error: any) {
+    this._isSearching = false;
+    this._isErrorOnSearching = true;
+    this._lastErrorMessage = 'При пошуку виникла помилка: \r\n' + <any>error;
```

Comment:
This message shows then when user don't found.
For this error user shoud see smth liek this:'User with these parameters  isn't found.'